example/grpc/middleware/tracing/tracing.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+// gRPC tracing middleware
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-rpc.md
+import (
+	"context"
+	"log"
+
+	"go.opentelemetry.io/otel/plugin/grpctrace"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/distributedcontext"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/exporter/trace/stdout"
+	"go.opentelemetry.io/otel/global"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+func init() {
+	initOtelTracer()
+}
```

Comment:
Mind moving `initOtelTracer` out of `init`, e.g., into a helper function that can be called from the two `main` functions? My concern is that we don't want instrumentation to have initialization side effects like tracer or exporter construction, so doing so in an example might be misleading.

Reply:
Makes sense. On it.